developer/styleguide.md

```diff
@@ -54,66 +55,111 @@ In our Markdown documentation, we use [remark-lint](https://github.com/wooorm/re
 
 Pull request branches are evaluated using [BitHound](https://www.bithound.io/github/reactioncommerce/reaction) for insecure dependencies and code quality.
 
-- must pass a Linting check and *no errors* are accepted.
-- must pass a Dependency check and no priority packages are accepted.
-- must pass `reaction test`
+-   must pass a Linting check and _no errors_ are accepted.
+-   must pass a Dependency check and no priority packages are accepted.
+-   must pass `reaction test`
 
 In many cases, documentation updates can be required as well.
 
 Pull requests are submitted to a peer code review process before acceptance.
 
-### File Naming Conventions
+### Naming Conventions
 
-In general we use hyphens (-) and camelCase for folder names, and camelCase alone for file names. Underscores are not to be used for file or folder names unless expressly required.  Be aware that not all operating systems are case sensitive, so it's not ok to have two files named the same with differing case.
+In general we use hyphens (-) for folder names. Underscores are not to be used for file or folder names unless expressly required.  Be aware that not all operating systems are case sensitive, so it's not ok to have two files named the same with differing case. To prevent this, we recommend not using camelCasing when naming folders or files.
 
-#### Folder Names
+As a convention, if there are multiple files that provide functionality that is broken down across multiple files, these files should be grouped within a folder. If the files are stand-alone and the name needs to represent functionality, you can use a hyphen for the seperator.
 
 **Good**
 
-- Folder names for packages must contain only lowercased alpha numeric characters and may be hyphenated if joining more than one word
-- Folder names all normal directories must start with a lowercased letter and may camel cased if joining more than one word
+```sh
+/container/settings.js
+/container/settings.html
+/container/ <more files>
+```
+
+or
+
+```sh
+/settings-container.js
+/settings-container.html
+```
+
+**Bad**
+
+```sh
+/container/settings-containers.js
+/container/settingsContainer.html
+settingsContainer.js
+```
+
+#### Folders
+
+**Good**
 
+-   Folder names for packages should contain only lowercased alpha numeric characters and are **hyphenated** if joining more than one word
 
+```sh
     // Packages in /imports/plugins
     ui-grid/
     example-paymentmethod/
     social/
     taxes-avalara/
-
-    // All other folder names everywhere
-    addressBook/
+```
 
 **Bad**
 
-- Package name should contain hyphens to make it easier to read.
-- Underscores are not to be used unless expressly required.
-
+-   Package name should contain hyphens to make it easier to read.
+-   Underscores are not to be used unless expressly required.
+-   Folder names should not be camelCased.
```

Comment:
Should we have a refactor party to get everything updated to hyphens instead of camelcase?

Reply:
![image](https://user-images.githubusercontent.com/4482263/29884252-b3a999d0-8d67-11e7-8348-83a42efd956a.png)